### DISCUSSION TITLE
[General] Deploy Mainnet Blockchain Based zkSync

### DISCUSSION THREAD

**Discussion by avenbreaks** [zkSync-Community-Hub/zksync-developers#488] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

How and which documentation should we follow to build a mainnet based on zkSync Hyperchains? We can build for testnet but the production documentation is still under development and directed to third party RaaS.

we have baremetal server to build Mainnet based on zkSync. 

### What is the specific issue or error you're encountering?

Development Mainnet Based zkSync on Baremetal

### Can you share the error messages or logs you're receiving, if any?

Null

### Have you made any recent changes to the contract before encountering this issue?

Null

### Are there any external libraries or contracts that your contract interacts with?

Null

### Can you provide the relevant portions of your contract code where the issue is occurring?

Null

### Have you tried to isolate the problem, and if so, what were the results?

Null

### What steps have you already taken to try to resolve the issue?

Null

**Comment by bxpana** [comment]
Appreciate the post. Will take some time to check it out later today

**Comment by githubdoramon** [comment, marked as answer]
@avenbreaks current ZK Stack configuration is not yet ready for production chains (but we are really close to it). Do you mind sending more information about your project to zkstack@matterlabs.dev so we can talk more about next steps?
